src/Storages/MergeTree/BackgroundJobsAssignee.h

```diff
@@ -81,11 +64,28 @@ class BackgroundJobsAssignee : public WithContext
         ContextPtr global_context_);
 
 private:
+    MergeTreeData & data;
```

Comment:
Moved below to initialize `type` in the first place.
Another options: 
1) pass `type_` to the `getSettings()`
2) Pass settings to the constructor (but we already have a context accessible)